diff --git a/CMakeLists.txt b/CMakeLists.txt
index 37c66b8..a848838 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,6 +1,6 @@
 # cmake project file by Prakash Punnoor
 # improved by Ryan Pavlik
-cmake_minimum_required(VERSION 2.6)
+cmake_minimum_required(VERSION 2.8.12)
 
 project(Alut C)
 
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index e24bcb6..9c2b3dd 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -39,11 +39,12 @@ set(ALUT_HEADERS
 	../include/AL/alut.h)
 source_group(APIHeader FILES ${ALUT_HEADERS})
 
-
 if(BUILD_STATIC)
 	add_library(alut_static STATIC ${ALUT_SOURCES} ${ALUT_INTERNAL_HEADERS} ${ALUT_HEADERS})
 	target_compile_definitions(alut_static PUBLIC ALUT_BUILD_STATIC)
 	target_link_libraries(alut_static ${OPENAL_LIBRARY} ${ADD_LIBS})
+	target_include_directories(alut_static PUBLIC ${OPENAL_INCLUDE_DIR})
+	set_property(TARGET alut_static PROPERTY PUBLIC_HEADER ${ALUT_HEADERS})
 	if(UNIX)
 		target_link_libraries(alut_static m)
 	endif()
@@ -61,6 +62,7 @@ if(BUILD_DYNAMIC)
 	endif()
 	set_property(TARGET alut PROPERTY PUBLIC_HEADER ${ALUT_HEADERS})
 	set_target_properties(alut PROPERTIES VERSION ${MAJOR_VERSION} SOVERSION ${MAJOR_VERSION})
+	target_include_directories(alut PUBLIC ${OPENAL_INCLUDE_DIR})
 	target_link_libraries(alut ${OPENAL_LIBRARY})
 	if(UNIX)
 		target_link_libraries(alut m)
@@ -69,8 +71,32 @@ if(BUILD_DYNAMIC)
 endif()
 
 install(TARGETS ${TARGET_LIBS}
+	EXPORT "FreeALUTTargets"
 	LIBRARY DESTINATION lib${LIB_SUFFIX}
 	ARCHIVE DESTINATION lib${LIB_SUFFIX}
 	RUNTIME DESTINATION bin
-	PUBLIC_HEADER DESTINATION include/AL)
+	PUBLIC_HEADER DESTINATION include/AL
+	)
+
+include(CMakePackageConfigHelpers)
+
+set(version_config "${CMAKE_CURRENT_BINARY_DIR}/temp/FreeALUTConfigVersion.cmake")
+set(project_config "${CMAKE_CURRENT_BINARY_DIR}/temp/FreeALUTConfig.cmake")
+set(namespace "FreeALUT::")
+
+write_basic_package_version_file("${version_config}"
+									COMPATIBILITY SameMajorVersion
+									VERSION ${PACKAGE_VERSION})
+
+configure_package_config_file("Config.cmake.in"
+								"${project_config}"
+								INSTALL_DESTINATION
+								lib${LIB_SUFFIX}/cmake/freealut)
+
+install(FILES "${project_config}" "${version_config}"
+		DESTINATION lib${LIB_SUFFIX}/cmake/freealut)
+
+install(EXPORT "FreeALUTTargets"
+		NAMESPACE "${namespace}"
+		DESTINATION lib${LIB_SUFFIX}/cmake/freealut)
 
diff --git a/src/Config.cmake.in b/src/Config.cmake.in
new file mode 100644
index 0000000..2a20f59
--- /dev/null
+++ b/src/Config.cmake.in
@@ -0,0 +1,5 @@
+
+@PACKAGE_INIT@
+
+include("${CMAKE_CURRENT_LIST_DIR}/FreeALUTTargets.cmake")
+check_required_components("FreeALUT")
